cpp/oneapi/dal/algo/connected_components/backend/cpu/vertex_partitioning_default_kernel.hpp

```diff
@@ -20,16 +20,157 @@
+    const auto max_vertex_id = dal::detail::integral_cast<std::int32_t>(vertex_count - 1);
+    for (std::int32_t u = 0; u <= max_vertex_id; ++u) {
+        if (components[u] == u) {
+            components[u] = ordered_comp_id;
+            component_count++;
+            ordered_comp_id++;
+        }
+        else {
+            components[u] = components[components[u]];
+        }
+    }
+}
+
+template <typename Cpu>
+inline std::int32_t most_frequent_element(const std::int32_t *components,
+                                          const std::int64_t &vertex_count,
+                                          byte_alloc_iface *alloc_ptr,
+                                          const std::int32_t &samples_num = 1024) {
+    using vertex_type = std::int32_t;
+    using vertex_allocator_type = inner_alloc<vertex_type>;
+
+    vertex_allocator_type vertex_allocator(alloc_ptr);
+
+    vertex_type *rnd_vertex_ids = allocate(vertex_allocator, samples_num);
+
+    dal::backend::primitives::engine eng;
+    dal::backend::primitives::rng<std::int32_t> rn_gen;
+    rn_gen.uniform(samples_num, rnd_vertex_ids, eng.get_state(), 0, vertex_count);
+
+    vertex_type *sample_counts = allocate(vertex_allocator, vertex_count);
+    const auto max_vertex_id = dal::detail::integral_cast<std::int32_t>(vertex_count - 1);
+    for (std::int32_t u = 0; u <= max_vertex_id; ++u) {
+        sample_counts[u] = 0;
+    }
+    for (std::int32_t i = 0; i < samples_num; ++i) {
+        sample_counts[components[rnd_vertex_ids[i]]]++;
+    }
+    deallocate(vertex_allocator, rnd_vertex_ids, samples_num);
+
+    std::int32_t max_sample_count = 0;
+    std::int32_t most_frequent_root = 0;
+    for (std::int32_t u = 0; u <= max_vertex_id; ++u) {
+        if (sample_counts[u] > max_sample_count) {
+            max_sample_count = sample_counts[u];
+            most_frequent_root = u;
+        }
+    }
+    deallocate(vertex_allocator, sample_counts, vertex_count);
+
+    return most_frequent_root;
+}
 
 template <typename Cpu>
 struct afforest {
     vertex_partitioning_result<task::vertex_partitioning> operator()(
-        const detail::descriptor_base<task::vertex_partitioning>& desc,
-        const dal::preview::detail::topology<std::int32_t>& t) {
-        throw unimplemented(oneapi::dal::detail::error_messages::method_not_implemented());
+        const detail::descriptor_base<task::vertex_partitioning> &desc,
+        const dal::preview::detail::topology<std::int32_t> &t,
+        byte_alloc_iface *alloc_ptr) {
+        using vertex_type = std::int32_t;
+        using vertex_allocator_type = inner_alloc<vertex_type>;
+
+        vertex_allocator_type vertex_allocator(alloc_ptr);
+
+        const auto vertex_count = t.get_vertex_count();
+        const auto max_vertex_id = dal::detail::integral_cast<std::int32_t>(vertex_count - 1);
+
+        vertex_type *components = allocate(vertex_allocator, vertex_count);
+
+        for (std::int32_t u = 0; u <= max_vertex_id; ++u) {
```

Comment:
if you have started to use alias - vertex_type - follow its usage than vertex type shall be used in the code. Or why did you introduce it?